src/cpu.h

```diff
@@ -16,9 +16,73 @@
 #define NCNN_CPU_H
 
 #include <stddef.h>
+#if defined __ANDROID__ || defined __linux__
+#include <sched.h>
+#else
+#include <cstring>
+#endif
 
 namespace ncnn {
 
+class CpuSet
+{
+private:
+#if defined __ANDROID__ || defined __linux__
+    cpu_set_t m_bits;
+#else
+#define CPU_SETSIZE   1024
+#define NCNN_NCPUBITS (8 * sizeof(unsigned long))
+    unsigned long m_bits[CPU_SETSIZE / NCNN_NCPUBITS];
+#endif
+
+public:
+    void* data_ptr()
+    {
+        return static_cast<void*>(&m_bits);
+    }
+
+    size_t data_size()
+    {
+        return sizeof(m_bits);
+    }
```

Comment:
These two functions are implementation-specific, and for internal use only.
Move them to protected block, and add related friend functions